// @hashgraph/sdk/lib/index-web.js
"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
const BaseClient_1 = require("./BaseClient");
const grpc_web_1 = require("@improbable-eng/grpc-web");
var Code = grpc_web_1.grpc.Code;
__export(require("./exports"));
const mainnetProxy = { "https://grpc-web.myhbarwallet.com": { shard: 0, realm: 0, account: 3 } };
const testnetProxy = { "https://grpc-web.testnet.myhbarwallet.com": { shard: 0, realm: 0, account: 3 } };
/** This implementation of `BaseClient` is exported for browser usage. */
class Client extends BaseClient_1.BaseClient {
    /**
     * If `network` is not specified, default url is a proxy to 0.testnet.hedera.com:50211 generously
     * hosted by MyHbarWallet.com. Mainnet proxy to come later.
     */
    constructor({ network = testnetProxy, operator }) {
        super(network, operator);
    }
    static forMainnet() {
        return new Client({ network: mainnetProxy });
    }
    static forTestnet() {
        return new Client({ network: testnetProxy });
    }
    static fromFile() {
        throw new Error("Client.fromFile is not supported in the browser");
    }
    static fromJson(text) {
        return new Client(JSON.parse(text));
    }
    close() {
        throw new Error("Client.close is not supported in the browser");
    }
    /* eslint-disable-next-line @typescript-eslint/member-naming */
    _unaryCall(url, request, method) {
        return new Promise((resolve, reject) => grpc_web_1.grpc.unary(method, {
            host: url,
            request,
            onEnd(response) {
                if (response.status === Code.OK && response.message != null) {
                    resolve(response.message);
                }
                else {
                    reject(new Error(response.statusMessage));
                }
            }
        }));
    }
}
exports.Client = Client;
// Mirror
var MirrorClient_1 = require("./mirror/web/MirrorClient");
exports.MirrorClient = MirrorClient_1.MirrorClient;
var MirrorConsensusTopicQuery_1 = require("./mirror/web/MirrorConsensusTopicQuery");
exports.MirrorConsensusTopicQuery = MirrorConsensusTopicQuery_1.MirrorConsensusTopicQuery;
