// @hashgraph/sdk/lib/generated/ConsensusTopicState_pb.js
// source: ConsensusTopicState.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var Timestamp_pb = require('./Timestamp_pb.js');
goog.object.extend(proto, Timestamp_pb);
goog.exportSymbol('proto.proto.ConsensusTopicState', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.ConsensusTopicState = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.proto.ConsensusTopicState, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.ConsensusTopicState.displayName = 'proto.proto.ConsensusTopicState';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.ConsensusTopicState.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.ConsensusTopicState.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.ConsensusTopicState} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.ConsensusTopicState.toObject = function(includeInstance, msg) {
  var f, obj = {
    sequencenumber: jspb.Message.getFieldWithDefault(msg, 1, 0),
    runninghash: msg.getRunninghash_asB64(),
    lastsubmittime: (f = msg.getLastsubmittime()) && Timestamp_pb.Timestamp.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.ConsensusTopicState}
 */
proto.proto.ConsensusTopicState.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.ConsensusTopicState;
  return proto.proto.ConsensusTopicState.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.ConsensusTopicState} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.ConsensusTopicState}
 */
proto.proto.ConsensusTopicState.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setSequencenumber(value);
      break;
    case 2:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setRunninghash(value);
      break;
    case 3:
      var value = new Timestamp_pb.Timestamp;
      reader.readMessage(value,Timestamp_pb.Timestamp.deserializeBinaryFromReader);
      msg.setLastsubmittime(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.ConsensusTopicState.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.ConsensusTopicState.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.ConsensusTopicState} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.ConsensusTopicState.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSequencenumber();
  if (f !== 0) {
    writer.writeUint64(
      1,
      f
    );
  }
  f = message.getRunninghash_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      2,
      f
    );
  }
  f = message.getLastsubmittime();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      Timestamp_pb.Timestamp.serializeBinaryToWriter
    );
  }
};


/**
 * optional uint64 sequenceNumber = 1;
 * @return {number}
 */
proto.proto.ConsensusTopicState.prototype.getSequencenumber = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.proto.ConsensusTopicState} returns this
 */
proto.proto.ConsensusTopicState.prototype.setSequencenumber = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional bytes runningHash = 2;
 * @return {!(string|Uint8Array)}
 */
proto.proto.ConsensusTopicState.prototype.getRunninghash = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * optional bytes runningHash = 2;
 * This is a type-conversion wrapper around `getRunninghash()`
 * @return {string}
 */
proto.proto.ConsensusTopicState.prototype.getRunninghash_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getRunninghash()));
};


/**
 * optional bytes runningHash = 2;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getRunninghash()`
 * @return {!Uint8Array}
 */
proto.proto.ConsensusTopicState.prototype.getRunninghash_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getRunninghash()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.proto.ConsensusTopicState} returns this
 */
proto.proto.ConsensusTopicState.prototype.setRunninghash = function(value) {
  return jspb.Message.setProto3BytesField(this, 2, value);
};


/**
 * optional Timestamp lastSubmitTime = 3;
 * @return {?proto.proto.Timestamp}
 */
proto.proto.ConsensusTopicState.prototype.getLastsubmittime = function() {
  return /** @type{?proto.proto.Timestamp} */ (
    jspb.Message.getWrapperField(this, Timestamp_pb.Timestamp, 3));
};


/**
 * @param {?proto.proto.Timestamp|undefined} value
 * @return {!proto.proto.ConsensusTopicState} returns this
*/
proto.proto.ConsensusTopicState.prototype.setLastsubmittime = function(value) {
  return jspb.Message.setWrapperField(this, 3, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.ConsensusTopicState} returns this
 */
proto.proto.ConsensusTopicState.prototype.clearLastsubmittime = function() {
  return this.setLastsubmittime(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.ConsensusTopicState.prototype.hasLastsubmittime = function() {
  return jspb.Message.getField(this, 3) != null;
};


goog.object.extend(exports, proto.proto);
