// @hashgraph/sdk/lib/generated/ConsensusTopicDefinition_pb.js
// source: ConsensusTopicDefinition.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var BasicTypes_pb = require('./BasicTypes_pb.js');
goog.object.extend(proto, BasicTypes_pb);
var Timestamp_pb = require('./Timestamp_pb.js');
goog.object.extend(proto, Timestamp_pb);
goog.exportSymbol('proto.proto.ConsensusTopicDefinition', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.ConsensusTopicDefinition = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.proto.ConsensusTopicDefinition, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.ConsensusTopicDefinition.displayName = 'proto.proto.ConsensusTopicDefinition';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.ConsensusTopicDefinition.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.ConsensusTopicDefinition.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.ConsensusTopicDefinition} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.ConsensusTopicDefinition.toObject = function(includeInstance, msg) {
  var f, obj = {
    memo: jspb.Message.getFieldWithDefault(msg, 1, ""),
    validstarttime: (f = msg.getValidstarttime()) && Timestamp_pb.Timestamp.toObject(includeInstance, f),
    expirationtime: (f = msg.getExpirationtime()) && Timestamp_pb.Timestamp.toObject(includeInstance, f),
    adminkey: (f = msg.getAdminkey()) && BasicTypes_pb.Key.toObject(includeInstance, f),
    submitkey: (f = msg.getSubmitkey()) && BasicTypes_pb.Key.toObject(includeInstance, f),
    lastupdatetime: (f = msg.getLastupdatetime()) && Timestamp_pb.Timestamp.toObject(includeInstance, f),
    creationtime: (f = msg.getCreationtime()) && Timestamp_pb.Timestamp.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.ConsensusTopicDefinition}
 */
proto.proto.ConsensusTopicDefinition.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.ConsensusTopicDefinition;
  return proto.proto.ConsensusTopicDefinition.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.ConsensusTopicDefinition} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.ConsensusTopicDefinition}
 */
proto.proto.ConsensusTopicDefinition.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setMemo(value);
      break;
    case 2:
      var value = new Timestamp_pb.Timestamp;
      reader.readMessage(value,Timestamp_pb.Timestamp.deserializeBinaryFromReader);
      msg.setValidstarttime(value);
      break;
    case 3:
      var value = new Timestamp_pb.Timestamp;
      reader.readMessage(value,Timestamp_pb.Timestamp.deserializeBinaryFromReader);
      msg.setExpirationtime(value);
      break;
    case 4:
      var value = new BasicTypes_pb.Key;
      reader.readMessage(value,BasicTypes_pb.Key.deserializeBinaryFromReader);
      msg.setAdminkey(value);
      break;
    case 5:
      var value = new BasicTypes_pb.Key;
      reader.readMessage(value,BasicTypes_pb.Key.deserializeBinaryFromReader);
      msg.setSubmitkey(value);
      break;
    case 6:
      var value = new Timestamp_pb.Timestamp;
      reader.readMessage(value,Timestamp_pb.Timestamp.deserializeBinaryFromReader);
      msg.setLastupdatetime(value);
      break;
    case 7:
      var value = new Timestamp_pb.Timestamp;
      reader.readMessage(value,Timestamp_pb.Timestamp.deserializeBinaryFromReader);
      msg.setCreationtime(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.ConsensusTopicDefinition.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.ConsensusTopicDefinition.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.ConsensusTopicDefinition} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.ConsensusTopicDefinition.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMemo();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getValidstarttime();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      Timestamp_pb.Timestamp.serializeBinaryToWriter
    );
  }
  f = message.getExpirationtime();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      Timestamp_pb.Timestamp.serializeBinaryToWriter
    );
  }
  f = message.getAdminkey();
  if (f != null) {
    writer.writeMessage(
      4,
      f,
      BasicTypes_pb.Key.serializeBinaryToWriter
    );
  }
  f = message.getSubmitkey();
  if (f != null) {
    writer.writeMessage(
      5,
      f,
      BasicTypes_pb.Key.serializeBinaryToWriter
    );
  }
  f = message.getLastupdatetime();
  if (f != null) {
    writer.writeMessage(
      6,
      f,
      Timestamp_pb.Timestamp.serializeBinaryToWriter
    );
  }
  f = message.getCreationtime();
  if (f != null) {
    writer.writeMessage(
      7,
      f,
      Timestamp_pb.Timestamp.serializeBinaryToWriter
    );
  }
};


/**
 * optional string memo = 1;
 * @return {string}
 */
proto.proto.ConsensusTopicDefinition.prototype.getMemo = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.proto.ConsensusTopicDefinition} returns this
 */
proto.proto.ConsensusTopicDefinition.prototype.setMemo = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional Timestamp validStartTime = 2;
 * @return {?proto.proto.Timestamp}
 */
proto.proto.ConsensusTopicDefinition.prototype.getValidstarttime = function() {
  return /** @type{?proto.proto.Timestamp} */ (
    jspb.Message.getWrapperField(this, Timestamp_pb.Timestamp, 2));
};


/**
 * @param {?proto.proto.Timestamp|undefined} value
 * @return {!proto.proto.ConsensusTopicDefinition} returns this
*/
proto.proto.ConsensusTopicDefinition.prototype.setValidstarttime = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.ConsensusTopicDefinition} returns this
 */
proto.proto.ConsensusTopicDefinition.prototype.clearValidstarttime = function() {
  return this.setValidstarttime(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.ConsensusTopicDefinition.prototype.hasValidstarttime = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional Timestamp expirationTime = 3;
 * @return {?proto.proto.Timestamp}
 */
proto.proto.ConsensusTopicDefinition.prototype.getExpirationtime = function() {
  return /** @type{?proto.proto.Timestamp} */ (
    jspb.Message.getWrapperField(this, Timestamp_pb.Timestamp, 3));
};


/**
 * @param {?proto.proto.Timestamp|undefined} value
 * @return {!proto.proto.ConsensusTopicDefinition} returns this
*/
proto.proto.ConsensusTopicDefinition.prototype.setExpirationtime = function(value) {
  return jspb.Message.setWrapperField(this, 3, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.ConsensusTopicDefinition} returns this
 */
proto.proto.ConsensusTopicDefinition.prototype.clearExpirationtime = function() {
  return this.setExpirationtime(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.ConsensusTopicDefinition.prototype.hasExpirationtime = function() {
  return jspb.Message.getField(this, 3) != null;
};


/**
 * optional Key adminKey = 4;
 * @return {?proto.proto.Key}
 */
proto.proto.ConsensusTopicDefinition.prototype.getAdminkey = function() {
  return /** @type{?proto.proto.Key} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.Key, 4));
};


/**
 * @param {?proto.proto.Key|undefined} value
 * @return {!proto.proto.ConsensusTopicDefinition} returns this
*/
proto.proto.ConsensusTopicDefinition.prototype.setAdminkey = function(value) {
  return jspb.Message.setWrapperField(this, 4, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.ConsensusTopicDefinition} returns this
 */
proto.proto.ConsensusTopicDefinition.prototype.clearAdminkey = function() {
  return this.setAdminkey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.ConsensusTopicDefinition.prototype.hasAdminkey = function() {
  return jspb.Message.getField(this, 4) != null;
};


/**
 * optional Key submitKey = 5;
 * @return {?proto.proto.Key}
 */
proto.proto.ConsensusTopicDefinition.prototype.getSubmitkey = function() {
  return /** @type{?proto.proto.Key} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.Key, 5));
};


/**
 * @param {?proto.proto.Key|undefined} value
 * @return {!proto.proto.ConsensusTopicDefinition} returns this
*/
proto.proto.ConsensusTopicDefinition.prototype.setSubmitkey = function(value) {
  return jspb.Message.setWrapperField(this, 5, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.ConsensusTopicDefinition} returns this
 */
proto.proto.ConsensusTopicDefinition.prototype.clearSubmitkey = function() {
  return this.setSubmitkey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.ConsensusTopicDefinition.prototype.hasSubmitkey = function() {
  return jspb.Message.getField(this, 5) != null;
};


/**
 * optional Timestamp lastUpdateTime = 6;
 * @return {?proto.proto.Timestamp}
 */
proto.proto.ConsensusTopicDefinition.prototype.getLastupdatetime = function() {
  return /** @type{?proto.proto.Timestamp} */ (
    jspb.Message.getWrapperField(this, Timestamp_pb.Timestamp, 6));
};


/**
 * @param {?proto.proto.Timestamp|undefined} value
 * @return {!proto.proto.ConsensusTopicDefinition} returns this
*/
proto.proto.ConsensusTopicDefinition.prototype.setLastupdatetime = function(value) {
  return jspb.Message.setWrapperField(this, 6, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.ConsensusTopicDefinition} returns this
 */
proto.proto.ConsensusTopicDefinition.prototype.clearLastupdatetime = function() {
  return this.setLastupdatetime(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.ConsensusTopicDefinition.prototype.hasLastupdatetime = function() {
  return jspb.Message.getField(this, 6) != null;
};


/**
 * optional Timestamp creationTime = 7;
 * @return {?proto.proto.Timestamp}
 */
proto.proto.ConsensusTopicDefinition.prototype.getCreationtime = function() {
  return /** @type{?proto.proto.Timestamp} */ (
    jspb.Message.getWrapperField(this, Timestamp_pb.Timestamp, 7));
};


/**
 * @param {?proto.proto.Timestamp|undefined} value
 * @return {!proto.proto.ConsensusTopicDefinition} returns this
*/
proto.proto.ConsensusTopicDefinition.prototype.setCreationtime = function(value) {
  return jspb.Message.setWrapperField(this, 7, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.ConsensusTopicDefinition} returns this
 */
proto.proto.ConsensusTopicDefinition.prototype.clearCreationtime = function() {
  return this.setCreationtime(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.ConsensusTopicDefinition.prototype.hasCreationtime = function() {
  return jspb.Message.getField(this, 7) != null;
};


goog.object.extend(exports, proto.proto);
